src/app/components/core/scheduler/update-schedule/update-schedule.component.ts

```diff
@@ -94,71 +93,70 @@ export class UpdateScheduleComponent implements OnInit, OnChanges {
-    } else {
-      selectedDay = this.days[index - 1];
-    }
-    return selectedDay;
+  public setDayByIndex(index: number) {
+    // if day is null
+    index = index ? index : 0;
+    const day = this.days.find(day => day.index == index);
+    this.form.controls['day'].patchValue(day);
+    this.form.controls['day'].updateValueAndValidity();
+    // set default time 
+    this.form.controls['time'].patchValue('00:00:01');
+    this.form.controls['time'].updateValueAndValidity();
+    return day;
   }
 
-  /**
-   * getSelectedDay
-   */
-  public getSelectedDayIndex(day) {
-    this.selectedDayName = day;
-    return this.days.indexOf(day) + 1;
+  setDay(day: string) {
+    this.form.controls['day'].patchValue(day);
+    this.form.controls['day'].updateValueAndValidity();
   }
 
   /**
    * Get schedule
    * @param id to get schedule
    */
-  public getSchedule(id): void {
-    if (id === undefined) {
-      return;
-    }
-    this.schedulesService.getSchedule(id).
-      subscribe(
-        (data) => {
-          if (data['type'] === 'TIMED') {
-            this.selectedScheduleTypeIndex = this.setScheduleTypeKey(data['type']);
-            this.selectedDayName = this.getSelectedDay(data['day']);
-          } else {
-            this.selectedScheduleTypeIndex = this.setScheduleTypeKey(data['type']);
-          }
-
-          const repeatTime = Utils.secondsToDhms(data['repeat']);
-          const time = Utils.secondsToDhms(data['time']);
-
-          // used for enable / disable switch only
-          this.scheduleName = data['name'].replace(' ', '');
-          // Fill form field values
+  public getSchedule(id: string): void {
+    this.schedulesService.getSchedule(id)
+      .subscribe(
+        (schedule: Schedule) => {
+          const repeatTime = Utils.secondsToDhms(schedule.repeat);
+          const time = Utils.secondsToDhms(schedule.time);
           this.form.patchValue({
-            name: data['name'],
+            name: schedule.name,
+            type: this.scheduleTypes?.find(t => t.name === schedule.type),
             repeatDay: repeatTime.days,
             repeat: repeatTime.time,
-            exclusive: data['exclusive'],
-            process_name: data['processName'],
-            day: this.selectedDayName,
+            exclusive: schedule.exclusive,
+            processName: schedule.processName,
+            day: schedule.type == 'TIMED' ? this.setDayByIndex(schedule.day) : 0,
             time: time.time,
-            enabled: data['enabled']
+            enabled: schedule.enabled
           });
+
+          this.schedule = cloneDeep(this.form.getRawValue());
+          this.toggleModal(true);
         },
         error => {
           if (error.status === 0) {
             console.log('service down ', error);
           } else {
-            this.alertService.error(error.statusText);
+            this.alertService.error(error.statusText, true);
           }
+          this.navToSchedulesPage();
```

Comment:
even in the case of error?

Reply:
in case of invalid scheduleId or any error, an error message will be shown and modal will be closed and user will be redirected on the list page.